java/test/TestH5A.java

```diff
@@ -1863,4 +1870,134 @@ public void testH5AArraywr()
+        arr_str_data[3]          = new ArrayList<String>(Arrays.asList(str_data3));
+        arr_str_data[4]          = new ArrayList<String>(Arrays.asList(str_data4));
+        arr_str_data[5]          = new ArrayList<String>(Arrays.asList(str_data5));
+
+        try {
+            H5atid = H5.H5Tcopy(HDF5Constants.H5T_C_S1);
+        }
+        catch (Throwable err) {
+            err.printStackTrace();
+            fail("testH5AArray_string_buffer.H5.H5Tcopy: " + err);
+        }
+        assertTrue("testH5AArray_string_buffer.H5Tcopy: ", H5atid >= 0);
+        try {
+            H5.H5Tset_size(H5atid, HDF5Constants.H5T_VARIABLE);
+            assertTrue("testH5AArray_string_buffer.H5Tis_variable_str", H5.H5Tis_variable_str(H5atid));
+        }
+        catch (Throwable err) {
+            err.printStackTrace();
+            fail("testH5DArray_string_buffer.H5Tset_size: " + err);
+        }
+        try {
+            atype_str_id = H5.H5Tarray_create(H5atid, 1, strdims);
+            assertTrue("testH5AArray_string_buffer.H5Tarray_create: ", atype_str_id >= 0);
+        }
+        catch (Exception err) {
+            if (atype_str_id > 0)
+                try {
+                    H5.H5Tclose(atype_str_id);
+                }
+                catch (Exception ex) {
+                }
+            err.printStackTrace();
+            fail("testH5AArray_string_buffer: " + err);
+        }
+
+        try {
+            aspace_id = H5.H5Screate_simple(1, dims, null);
+            assertTrue(aspace_id > 0);
+            att_str_id = H5.H5Acreate(H5did, att_str_name, atype_str_id, aspace_id, HDF5Constants.H5P_DEFAULT,
+                                      HDF5Constants.H5P_DEFAULT);
+            assertTrue("testH5AArray_string_buffer: ", att_str_id >= 0);
+
+            H5.H5AwriteVL(att_str_id, atype_str_id, arr_str_data);
+        }
+        catch (Exception err) {
+            if (att_str_id > 0)
+                try {
+                    H5.H5Dclose(att_str_id);
+                }
+                catch (Exception ex) {
+                }
+            if (atype_str_id > 0)
+                try {
+                    H5.H5Tclose(atype_str_id);
+                }
+                catch (Exception ex) {
+                }
+            err.printStackTrace();
+            fail("testH5AArray_string_buffer: " + err);
+        }
+        finally {
+            if (aspace_id > 0)
+                try {
+                    H5.H5Sclose(aspace_id);
+                }
+                catch (Exception ex) {
+                }
+        }
+
+        H5.H5Fflush(H5fid, HDF5Constants.H5F_SCOPE_LOCAL);
+
+        for (int j = 0; j < dims.length; j++)
+            lsize *= dims[j];
+
+        ArrayList[] arr_readbuf = new ArrayList[6];
+        for (int j = 0; j < lsize; j++)
+            arr_readbuf[j] = new ArrayList<String>();
+
+        try {
+            H5.H5AreadVL(att_str_id, atype_str_id, arr_readbuf);
```

Comment:
Are the read and write going to assume that the user is passing in an ArrayList[]? I can see that being problematic for several reasons.

Reply:
No only when it is a container type.